### ISSUE TITLE
Save entered texts as drafts instead of discarding them

### ISSUE THREAD

**Issue by Vampire** [k0shk0sh/FastHub#811]
**FastHub Version: 4.0.3**  
**Android Version: 7.1.2 (SDK: 25)**  
**Device Information:**  
- MANUFACTURER: samsung
- BRAND: samsung  
- MODEL: SM-G930F  
---

 It would be nice to be able to review a conversation before submitting a comment. If you are writing a comment and want to re-read what was written before to not forget anything, you have to copy all text, discard the changes, read the conversation, then reopen the dialogue and paste what you have copied and you have to make sure you didn't copy anything else in the meantime, or your text is lost. 

It would really be nice if you instead e. g. could write, then go back, having the text saved as draft and as soon as you go to commenting again, have the text presented. There could even be some point in the navigation that shows you your draft.

If you e. g. start to write some comment, that are distracted by another issue, start there to write, then your boss disturbs, ... 

For creating new issues it would even be nice to have the option to continue one of several drafts or to start a new draft. Then you could e. g. start to report some issues, but delay submission until more information is collected and added.

**Comment by k0shk0sh** [comment]
Isn't the reply button meant for this? It will show you the user reply in top so you could see the conversation? This issue has been opened before and the way FastHub did it was by adding the reply functionality. 
Anyway I'm thinking of another way of doing such improvement according to the discussion in the other issue, is by splitting the screen so you could see everything while typing.

_Sent from my Samsung SM-G950F using [FastHub Debug](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by k0shk0sh** [comment]
#438

_Sent from my Samsung SM-G950F using [FastHub Debug](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Vampire** [comment]
As far as I have seen the Reply button is only working for one comment, so you cannot review more than that single comment while answering. 

A split screen, or a left-right-swipable screen would for sure help for writing and could be done additionally to what I suggested here. 

But such a draft functionally could be handy for more than that. As I described, e. g.

- for preparing multiple issues in parallel
- for collecting more information before submitting an issue or comment
- when reporting a FastHub issue to test again what you are about to report before you submit it
- for not losing texts when you are disturbed by bus, work, wife, kids, ...
- ...

**Comment by k0shk0sh** [comment]
@Vampire  I understand the hassle I'll try my best to bring that feature to my top of my todos and I can't promise you but I'll try my best that it'll in next version ;)

_Sent from my Samsung SM-G950F using [FastHub Debug](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Vampire** [comment]
No stress, I have a workaround, so no problem. :-)
But what do you mean, the split screen or the drafting? 
If the latter, then this should be reopened.
If the first, it would still be cool if you could consider the drafting. :-)

**Comment by k0shk0sh** [comment]
You wont be needing draft, as the whole idea behind draft is to able to see others while u commenting. 

_Sent from my Samsung SM-G950F using [FastHub Debug](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by yakov116** [comment]
@k0shk0sh really quick commenting will solve the issue (adding on the bottom like there is not for adding a comment in a pr)

_Sent from my Google Nexus 9 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Vampire** [comment]
@k0shk0sh no it is not, at least not for me. It was the initial intent, that made me start reporting this issue. But as I wrote it and we talked, I have seen other use-cases where it would be handy.

Let me elaborate again on some of them:

- You would be able to prepare multiple issues in parallel on the same or even on different projects. This would have been very helpful e. g. when I reported the nested formatting issue. While trying out and writing the text, I found several other issues that I would have wanted to report right away. Now I had to not them down separately to report them later on, maybe forgetting some details until I create tickets for them.
- You could start reporting an issue, but collect more information before submitting the issue or comment. This would have been very helpful e. g. when I reported the nested formatting issue. I needed quite some time and effort to collect all oks and not-oks.
- While reporting a FastHub issue via FastHub, it can be helpful to test again what you are about to report before you submit it to be able to write better bug reports.
- When you are in the process of writing a comment or issue and you are disturbed by bus arriving, work, wife, kids, other app or whatever, you would be safe against losing texts you started to write if there is a draft saved as soon as the dialog is left or the activity paused.
- When you are in the middle of writing an issue or comment and there is some other important issue reported or commented on what you see my email for example, you could simply switch to that other issue write a comment or even spend three days for a high-priority bugfix and then later on come back to the text you started.
- Probably many other use-cases where this could be helpful.

All this and probably more is not solved by the split screen approach.

**Comment by bb010g** [comment]
I just lost a kinda long in-progress comment due to the lack of auto-saving drafts and FastHub deciding that I was done with that activity instead of doing its normal multiplexing in the activity switcher, so +1 for drafts from me, completely aside from looking at the previous comments in a thread.

Drafts are also nice for avoiding having to buffer the text outside of FastHub (clipboard, note app, wherever) when you want to pause the current comment you're composing, go quickly reply to a different issue, and then get back to the initial comment-in-progress.

**Comment by k0shk0sh** [comment]
Lets wait for v5 :)
